modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaSearchVocabulariesMethodCheck.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaSearchVocabulariesMethodCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		String newContent = content;
+
+		Matcher matcher = _searchVocabulariesPattern.matcher(content);
+
+		boolean replaced = false;
```

Comment:
@TamyrisTorres, please keep the declaration of the variable `matcher` close to the loop `while`. I mean:
```
boolean replaced = false;

Matcher matcher = _searchVocabulariesPattern.matcher(content);

while (matcher.find()) {
```
In Liferay code, it's preferable to declare the variable being used to control a loop just before the loop. :wink: